src/intersections/raysegment.jl

```diff
@@ -0,0 +1,75 @@
+# ------------------------------------------------------------------
+# Licensed under the MIT License. See LICENSE in the project root.
+# ------------------------------------------------------------------
+
+#=
+The intersection type can be one of five types:
+
+1. intersect at one inner point (CrossingRaySegment -> Point)
+2. intersect at one corner point of segment xor origin of ray (MidTouchingRaySegment -> Point)
+3. intersects at one corner point of segment and origin of ray (CornerTouchingRaySegment -> Point)
+4. overlap at more than one point (OverlappingRaySegment -> Segment)
+5. do not overlap nor intersect (NoIntersection)
+=#
+function intersection(f, r1::Ray{N,T}, s1::Segment{N,T}) where {N,T}
+  a, b = r1(0), r1(1)
+  c, d = s1(0), s1(1)
+
+  λ₁, λ₂, r, rₐ = intersectparameters(a, b, c, d)
+
+  # not in same plane or parallel
+  if r ≠ rₐ
+    return @IT NoIntersection nothing f # CASE 5
```

Comment:
This line is not covered by tests.

Reply:
Skew lines are covered by a test in the 3D section, I add the parallel case with `s_{16}`:
![image](https://user-images.githubusercontent.com/67096719/191243068-f436830f-b21c-41b8-932a-fc3cebdfb2c9.png)

Reply:
@dorn-gerhard btw, it would be nice to add references to papers you are using to implement these methods. They are very useful for future maintainers to fix possible issues with so many branches. Thanks for the great work!